logging/access.go

```diff
@@ -109,9 +113,19 @@ func stripQueryString(u string) string {
 	}
 }
 
+// seedQueryParamMaskingHash hashes a provided query parameter value and returns the hashed value.
+// Uses the murmur3 algorithm and a 32-bit hasher set.
+func hashQueryParamValue(val string) string {
+	h32 := murmur3.New32WithSeed(seedQueryParamMaskingHash)
+	h32.Write([]byte(val))
+	hashedVal := h32.Sum32()
+
+	return fmt.Sprint(hashedVal)
+}
+
 // Logs an access event in Apache combined log format (with a minor customization with the duration).
 // Additional allows to provide extra data that may be also logged, depending on the specific log format.
-func LogAccess(entry *AccessEntry, additional map[string]interface{}) {
+func LogAccess(entry *AccessEntry, additional map[string]interface{}, maskedQueryParams []string) {
```

Comment:
There is an exported statebag key AccessLogAdditionalDataKey to provide additional data
and additional data can override predefined fields https://github.com/zalando/skipper/blob/9b570970ea935053830703fcd947b0aa76896625/logging/access.go#L152-L171

so another option is to create a filter that would mask request uri and put it into `ctx.StateBag()[al.AccessLogAdditionalDataKey]["uri"]`.

Reply:
Thank you for the feedback 👍

I updated the code and went with @szuecs's suggestion. Hope that's fine.

Reply:
Since we rely on additional data anyways lets use `"uri"` key - this way masking logic would belong to the filter and not to the access log package (e.g. there could be several ways to mask uri). The filter would need only to populate additional data in the statebag.

You can export `const UriAdditionalDataKey = "uri"` if you want and use it everywhere instead of `"uri"` .